src/notation/view/widgets/editstaff.cpp

```diff
@@ -492,6 +492,7 @@ void EditStaff::applyStaffProperties()
     config.clefTypeList = m_instrument.clefType(m_orgStaff->rstaff());
 
     notationParts()->setStaffConfig(m_orgStaff->id(), config);
+    m_orgStaff->setStaffType(Fraction(0, 1), editStaffTypeDialog->getStaffType());
```

Comment:
Unfortunately, here we need a fix of a different kind

1. we must change the staff type only by score()->undo(new Ms::ChangeStaffType(staff, newStaffType)) (i.e. in NotationParts::doSetStaffConfig). Otherwise it will be impossible to undo these changes (ctrl + z)

2. it seems that StaffConfig partially duplicates Ms::StaffType. I would suggest removing these duplicated fields from StaffConfig and adding Ms::StaffType as its member

Thus, we can pass the changed staff type directly to NotationParts and apply it using score()->undo(new Ms::ChangeStaffType(...))

Reply:
My apologies, thought the solution looked a bit naive!  Thanks for the guidance, I've implemented that.

Reply:
Thank you! Much better now! But you should rebase your PR: there should be only one commit